sscanss/core/instrument/create.py

```diff
@@ -208,7 +211,8 @@ def read_detector_description(instrument_data, positioners, path=''):
 
 
 def read_visuals(visuals_data, path=''):
-    """Creates Mesh object from a visuals description
```

Comment:
A lot of the validation is already done by the schema and will never get called, please see my suggested refactor below
```def read_visuals(visuals_data, path=''):
    """Creates Mesh object from a visuals description using a file if a mesh path supplied, 
    else the specified primitive geometry is used

    :param visuals_data: visuals description
    :type visuals_data: Optional[Dict]
    :param path: directory of the instrument description file
    :type path: str
    :return: mesh
    :rtype: Optional[Mesh]
    """
    if visuals_data:
        pose = visuals_data.get('pose', DEFAULT_POSE)
        pose = matrix_from_pose(pose)
        mesh_colour = visuals_data.get('colour', DEFAULT_COLOUR)

        if visuals_data.get('mesh'):
            mesh = read_3d_model(pathlib.Path(path).joinpath(visuals_data.get('mesh', '')).as_posix())

        geometry_data = visuals_data['geometry']
        geom_type = geometry_data['type'].capitalize()

        if geom_type == VisualGeometry.Mesh.value:
            mesh_file = geometry_data['path']
            mesh = read_3d_model(pathlib.Path(path).joinpath(mesh_file).as_posix())

        elif geom_type == VisualGeometry.Box.value:
            x, y, z = geometry_data['size']
            mesh = create_cuboid(x, z, y)

        elif geom_type == VisualGeometry.Plane.value:
            x, y = geometry_data['size']
            mesh = create_plane(Plane.fromCoefficient(1, 1, 0, 0), x, y)

        elif geom_type == VisualGeometry.Sphere.value:
            radius = geometry_data['radius']
            mesh = create_sphere(radius)

        mesh.transform(pose)
        mesh.colour = Colour(*mesh_colour)
        return mesh
```